EduNLP/I2V/i2v.py

```diff
@@ -317,6 +323,66 @@ def from_pretrained(cls, name, model_dir=MODEL_DIR, *args, **kwargs):
                    tokenizer_kwargs=tokenizer_kwargs)
 
 
+class Elmo(I2V):
+    """
+    The model aims to transfer item and tokens to vector with Elmo.
+
+    Bases
+    -------
+    I2V
+
+    Parameters
+    -----------
+    tokenizer: str
+        the tokenizer name
+    t2v: str
+        the name of token2vector model
+    args:
+        the parameters passed to t2v
+    tokenizer_kwargs: dict
+        the parameters passed to tokenizer
+    pretrained_t2v: bool
+        True: use pretrained t2v model
+        False: use your own t2v model
+    kwargs:
+        the parameters passed to t2v
+
+    Returns
+    -------
+    i2v model: Elmo
+    """
+
+    def infer_vector(self, items, tokenize=True, return_tensors='pt', *args, **kwargs) -> tuple:
+        """
+        It is a function to switch item to vector. And before using the function, it is nesseary to load model.
+
+        Parameters
+        -----------
+        items: str or list
+            the text of question
+        tokenize:bool
+            True: tokenize the item
+        return_tensors: str
+            tensor type used in tokenizer
+        args:
+            the parameters passed to t2v
+        kwargs:
+            the parameters passed to t2v
+
+        Returns
+        --------
+        vector:list
+        """
+        inputs = self.tokenize(items, return_tensors=return_tensors,
+                               pad_to_max_length=False) if tokenize is True else items
+        return self.t2v(inputs, *args, **kwargs), self.t2v.infer_tokens(inputs, *args, **kwargs)
+
+    @classmethod
+    def from_pretrained(cls, name, model_dir=MODEL_DIR, *args, **kwargs):
+        return cls("pure_text", name, pretrained_t2v=True, model_dir=model_dir)
```

Comment:
You need to test Elmo or get_pretrained_i2v with 'elmo_pub_math' here, it seem that you need to provide path in tokenizer_kwargs to init the tokenizer.